xtask/src/profile.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{cli::profile::Params, workload};
+use anyhow::Result;
+use std::collections::VecDeque;
+use std::process::Command;
+
+pub fn profile(params: Params) -> Result<()> {
+    // check for samply to be installed
+    Command::new("sh")
+        .args(["-c", "command", "-v", "samply"])
```

Comment:
seems strange to pull samply in to this. why can't the user just run `samply` themselves?

Reply:
this feels like feature/scope creep. unless absolutely necessary I don't think we should merge this.

Reply:
Interesting points, I will respond in the reverse order:

1. The only one we need to care about profiling is nomt
Agree, I will remove the option anyway to specify other backends

2. we don't need a generalizable profiler for all DBs, just a benchmark
That's true, but the workflow logic created for benchmarks would be great to be reutilized to be able to execute tasks and profile them with samply

3. seems strange to pull samply in to this. why can't the user just run samply themselves?
Agree, it is strange, but probably only because we're doing it in Rust. If done with a simple .sh script, it would not have been so strange. Anyway, there are two reasons why I call `samply` from here: the first one is that workloads need to be set up before being executed, and it would not be great to sample also the initialization (for example, the flame chart would always be polluted by a huge amount of writes if the specified initial capacity is big). The second reason is just because I personally like one-command actions, so being able to just write `cargo run --profile ...` seems great, instead of something like `cargo run --prepare_backend ... && cargo build ... && sampl target/...`."

Finally, the most important thing we need to define is: should we have a different binary to profile things, or is it okay to include the execution of a workload inside the bench binary?

Reply:
> That's true, but the workflow logic created for benchmarks would be great to be reutilized to be able to execute tasks and profile them with samply

This is feature creep and adds maintenance burden.

> If done with a simple .sh script, it would not have been so strange

No, it still would be strange. The user can just call `samply ./my_script.sh`.

Reply:
> it would not be great to sample also the initialization (for example, the flame chart would always be polluted by a huge amount of writes if the specified initial capacity is big).

This is fair, but I would instead prefer to address it by allowing the user to separate the initialization of the database from the benchmark. This is already desired, so it would be trivial to 

```
./the_binary initialize some_params
samply ./the_binary benchmark some_params
```

Reply:
Super valid points, I like the opinion to avoid introducing complexity! Separate initialization from execution in the cli would be very simple!